Adafruit Feather nrf52840 updates

jacobw opened the pull request:
The Adafruit Feather nrf52840 board was changed to Adafruit Feather recently, probably as part of the automated hwmv2 changes. The first commit gives the board the more specific name Adafruit Feather nrf52840 Express, and changes the directory to support other versions.

This second commit then adds the Adafruit Feather nrf52840 Sense, which is based on the same board but has more sensors, by making a common dts which is mostly shared.

jacobw committed:
boards: rename adafruit_feather

Adafruit feather is a generic name that could match many
boards. Renamed to reflect actual board name:
- Adafruit Feather nrf52840 Express

There are two boards based on the nrf52840, so keeping directory
generic to support additional board defs with mostly shared dts.

Signed-off-by: Jacob Winther <jacob@9.nz>

jacobw committed:
boards: Add Adafruit Feather nrf52840 Sense

Add support for board based on the Express variant.
Board uses uf2 as it has no swd adaptor.

Signed-off-by: Jacob Winther <jacob@9.nz>

decsny commented on boards/adafruit/feather_nrf52840/doc/index.rst:
do we need to add a code owner to this board? nobody is being added to review

jacobw replied:
Assume this isn’t a question for me?

decsny replied:
It's for you, or anybody who knows about this board, somebody should add themself to the code owner file or something to get pulled in to review if they have an interest in this board

jacobw replied:
I’m happy to own these boards if there is no one better to do it. I’ve been running my board defs out of tree for a while and there hasn’t been much other discussion on them as far as I’ve seen.

is there something specific I need to do?

decsny replied:
Just add your name to the code owner file in the top of the repo for these boards is what people usually do for stuff like this

You can do it as another commit in this PR

jacobw replied:
Ok will do this evening.

jacobw replied:
I added my name in another commit. Is there anything else I can do for this PR?

bjarki-andreasen replied:
@decsny There is a bit of a process to be added as maintainer/collaborator, see https://docs.zephyrproject.org/latest/project/project_roles.html

henrikbrixandersen replied:
@bjarki-andreasen There's a difference between the CODEOWNERS file and the MAINTAINERS.yaml file. The first does not require a process.

bjarki-andreasen replied:
@henrikbrixandersen
ok, I just read the "_This file is going to be deprecated and currently only had
entries that are not covered by the MAINTAINERS file._" section of the file, did not know we actually used it :)

decsny replied:
> @henrikbrixandersen ok, I just read the "_This file is going to be deprecated and currently only had entries that are not covered by the MAINTAINERS file._" section of the file, did not know we actually used it :)

yeah, I've seen that blurb, doesn't seem to be true, seems still the best place to put random board and driver owners and that's what I see people doing a lot

jacobw committed:
boards: Add code owner for feather_nrf52840

Add myself as owner for adafruit/feather_nrf52840 as there is
no current owner.

Discussed in PR 76135.

Signed-off-by: Jacob Winther <jacob@9.nz>

bjarki-andreasen commented on boards/adafruit/feather_nrf52840/Kconfig.adafruit_feather_nrf52840_sense:
```diff
@@ -0,0 +1,7 @@
+# Adafruit Feather nRF52840 Express board configuration
+
+# Copyright (c) 2020 Tobias Svehagen
```
The copyrights seem incorrect, this is a file created by Jacub right? copied from a file by Tobias Svehagen
In this case, add your own copyright to the file as well:
```
# Copyright (c) 2020 Tobias Svehagen
# Copyright (c) 2024 Jakub W
```

jacobw replied:
Yes that's correct. Ok I can amend as requested,

What's the threshold for adding a copyright? I made at least minor modifications to most of the files (i.e. in addition to renaming them). Should I amend all files I edited with a copyright?

bjarki-andreasen replied:
Don't know exactly :) I just want to be on the safe side

bjarki-andreasen reviewed (approved):
Looks good :) Nice work!

nordicjm reviewed (commented):
This is missing a deprecated board name entry, please add it otherwise anyone going from 3.7 or older will just have a "no such board" error

jacobw commented:
> This is missing a deprecated board name entry, please add it otherwise anyone going from 3.7 or older will just have a "no such board" error

PR #76624 created to resolve this.
